DigitalLearningSolutions.Web/Scripts/trackingSystem/emailDelegates.ts

```diff
@@ -0,0 +1,26 @@
+import { SearchSortFilterAndPaginate } from '../searchSortFilterAndPaginate/searchSortFilterAndPaginate';
+
+const selectedElements = document.querySelectorAll('.delegate-checkbox:checked') as NodeListOf<HTMLInputElement>;
+const selectedIds = Array.from(selectedElements).map((el) => el.value);
+const queryString = '?selectedIds='.concat(selectedIds.join('&selectedIds='));

```

Comment:
Any reason for "concat" over string interpolation? Also, how come we form a query string by having "selectedIds" as several different query strings

Reply:
This allows it to bind to a list.

Reply:
I see - there are some frameworks that would only select one value if a query string is repeated, rather than bind it into a list. I'd either use the `selectedIds[0]=1&selectedIds[1]=4` syntax, or write something custom to support comma separated format like `?selectedIds=1,2,4`

Unrelated, but why do we need the selectedIds when initially loading all the delegates with JS?

Reply:
This is the same way that the form data is sent with no-JS (as a bunch of selectedIds=xxx fields), and seems to be recommended online, so I went for that. But I've changed it to specify the indices as per your suggestion for clarity.

Whenever the page is "refreshed" e.g. when submitting the form causing a validation error, the page is returned with the previously selected ids still checked, as you would expect. Then the JS is "initially loaded" again, which overwrites the list of delegate checkboxes. So to catch this, whevener the JS is loaded, we need to get all the checked checkboxes and include them in the request when fetching the items so we get them prechecked appropriately. Of course, when the page is loaded for the very first time, there will be no selected checkboxes so the query param will be empty.

Reply:
That makes sense, thanks! 👍